geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/utils/CQEngineQueryVisitor.scala

```diff
@@ -0,0 +1,362 @@
+      case c => throw new RuntimeException(s"PropertyIsGreaterThan: $c not supported")
+    }
+  }
+
+  /**
+    * PropertyIsGreaterThanOrEqualTo
+    */
+  override def visit(filter: PropertyIsGreaterThanOrEqualTo, data: scala.Any): cqquery.simple.GreaterThan[SimpleFeature, _] = {
+    val prop = getProp(filter)
+    sft.getDescriptor(prop.name).getType.getBinding match {
+      case c if classOf[java.lang.Integer].isAssignableFrom(c) => BuildIntGTEQuery(prop)
+      case c if classOf[java.lang.Long   ].isAssignableFrom(c) => BuildLongGTEQuery(prop)
+      case c if classOf[java.lang.Float  ].isAssignableFrom(c) => BuildFloatGTEQuery(prop)
+      case c if classOf[java.lang.Double ].isAssignableFrom(c) => BuildDoubleGTEQuery(prop)
+      case c if classOf[java.util.Date   ].isAssignableFrom(c) => BuildDateGTEQuery(prop)
+      case c => throw new RuntimeException(s"PropertyIsGreaterThanOrEqualTo: $c not supported")
+    }
+  }
+
+  /**
+    * PropertyIsLessThan
+    */
+  override def visit(filter: PropertyIsLessThan, data: scala.Any): cqquery.simple.LessThan[SimpleFeature, _] = {
+    val prop = getProp(filter)
+    sft.getDescriptor(prop.name).getType.getBinding match {
+      case c if classOf[java.lang.Integer].isAssignableFrom(c) => BuildIntLTQuery(prop)
+      case c if classOf[java.lang.Long   ].isAssignableFrom(c) => BuildLongLTQuery(prop)
+      case c if classOf[java.lang.Float  ].isAssignableFrom(c) => BuildFloatLTQuery(prop)
+      case c if classOf[java.lang.Double ].isAssignableFrom(c) => BuildDoubleLTQuery(prop)
+      case c if classOf[java.util.Date   ].isAssignableFrom(c) => BuildDateLTQuery(prop)
+      case c => throw new RuntimeException(s"PropertyIsLessThan: $c not supported")
+    }
+  }
+
+  /**
+    * PropertyIsLessThanOrEqualTo
+    */
+  override def visit(filter: PropertyIsLessThanOrEqualTo, data: scala.Any): cqquery.simple.LessThan[SimpleFeature, _] = {
+    val prop = getProp(filter)
+    sft.getDescriptor(prop.name).getType.getBinding match {
+      case c if classOf[java.lang.Integer].isAssignableFrom(c) => BuildIntLTEQuery(prop)
+      case c if classOf[java.lang.Long   ].isAssignableFrom(c) => BuildLongLTEQuery(prop)
+      case c if classOf[java.lang.Float  ].isAssignableFrom(c) => BuildFloatLTEQuery(prop)
+      case c if classOf[java.lang.Double ].isAssignableFrom(c) => BuildDoubleLTEQuery(prop)
+      case c if classOf[java.util.Date   ].isAssignableFrom(c) => BuildDateLTEQuery(prop)
+      case c => throw new RuntimeException(s"PropertyIsLessThanOrEqualTo: $c not supported")
+    }
+  }
+
+  /**
+    * PropertyIsNotEqualTo
+    */
+  override def visit(filter: PropertyIsNotEqualTo, data: scala.Any): AnyRef = {
+    val prop = getProp(filter)
+    val attribute: Attribute[SimpleFeature, Any] = lookup.lookup[Any](prop.name)
+    val value: Any = prop.literal.evaluate(null, attribute.getAttributeType)
+    // TODO: could this be done better?
```

Comment:
Not too sure about leaving nebulous comments like this in. Thoughts? @jahhulbert-ccri @elahrvivaz @jnh5y